apps/website/src/components/content/YouTube.tsx

```diff
@@ -244,44 +219,22 @@ export const Lightbox = ({
     // Initialize the lightbox
     lightbox.init();
 
-    // Bind a custom lightbox event, so we can trigger the lightbox from a child
-    const cleanup = [
-      ...document.querySelectorAll<HTMLElement>(
-        `${opts.gallery} ${opts.children}`,
-      ),
-    ].map((elm) => {
-      const handler = (e: Event) => {
-        e.preventDefault();
-
-        // Create a custom event to pass to the lightbox
-        // Lightbox expects a MouseEvent from a listener bound to the gallery
-        const customEvent = new Event("lightbox");
-        Object.defineProperty(customEvent, "target", {
-          writable: false,
-          value: elm,
-        });
-        Object.defineProperty(customEvent, "currentTarget", {
-          writable: false,
-          value: document.querySelector(opts.gallery),
-        });
-        lightbox.onThumbnailsClick(customEvent as unknown as MouseEvent);
-      };
-      elm.addEventListener("lightbox", handler);
-      return () => {
-        elm.removeEventListener("lightbox", handler);
-      };
-    });
-
-    // Track that lightbox has been initialized
-    setInitialized(true);
-
     // Do the cleanup in the reverse order
     return () => {
-      setInitialized(false);
-      cleanup.forEach((fn) => fn());
       lightbox.destroy();
     };
-  }, [photoswipeId, updateConsent]);
+  }, [defaultOpen, photoswipeId, updateConsent]);
+
+  // Open the lightbox on load if we have a default
+  useEffect(() => {
+    if (defaultOpen && defaultOpen > -1) {
+      const gallery = document.getElementById(
+        photoswipeId,
+      ) as HTMLElement | null;
+      if (!gallery) return;
+      lightboxRef.current?.loadAndOpen(defaultOpen, { gallery });
+    }
+  }, [defaultOpen, photoswipeId]);
```

Comment:
We probably want to wait until the lightbox is initialized before doing this, and only handle if the useEffect for the lightbox gets re-run? The ref is probably fine, but we probably also still want a state to track once its initialized?

Reply:
I would've assumed the effects are done in series and they the init is synchronous, so it should be mostly guaranteed to be init'ed at that point?

Reply:
I would think so, but given they're different hooks with different dependencies, I would prefer to make this robust?

Reply:
sure, also i dont think your approach is bad. with this one the arrow buttons in the lightbox does not seem to work. its probably not as easy as i thought ;)